beacon-chain/db/kv/validated_checkpoint.go

```diff
@@ -19,7 +21,17 @@ func (s *Store) LastValidatedCheckpoint(ctx context.Context) (*ethpb.Checkpoint,
 		if enc == nil {
 			var finErr error
 			checkpoint, finErr = s.FinalizedCheckpoint(ctx)
-			return finErr
+			if finErr != nil {
+				return finErr
+			}
+			if bytes.Equal(checkpoint.Root, params.BeaconConfig().ZeroHash[:]) {
+				genesisRoot, genErr := s.GenesisBlockRoot(ctx)
```

Comment:
Does it matter we are opening a `s.db.View` inside a `s.db.View`?
You could just do:
```
		bkt = tx.Bucket(blocksBucket)
		r := bkt.Get(genesisBlockRootKey)
```